MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -831,20 +820,28 @@ namespace WPEFramework {
                         /* we set this to false */
                         g_is_critical_maintenance="false";
 
+
+                        if(m_thread.joinable()){
```

Comment:
I don't see a need for this. If the global/class variables are properly protected and notifications are sent appropriately.